cloud/docker-image/pom.xml

```diff
@@ -254,17 +254,89 @@
         <configuration>
           <filter>${*}</filter>
           <images>
```

Comment:
I think the main issue we need to figure out is how to control the merging of plugin configuration when release vs incubator profiles are activated.

By default, we have a single image built in the main pom, which could be the release or incubator images depending on local snapshot build vs performing the release build.

Then we want additional images to be included in the docker plugin configuration only for release, so those would be added in the release profile, but we need to use these maven hints to make sure they are merged as needed rather than replacing entirely.

See `advanced configuration inheritance` under https://maven.apache.org/pom.html#plugins, specifically `combine.self` and `combine.children`.

When testing the configuration, there's no need to execute the `docker-image` build, instead you can use the `help` plugin to see the effective pom observed by Maven.

```
./mvnw help:effective-pom
```
This will output the effective pom it uses to execute on the current project, which will let you try out different `combine.self` and `combine.children` hints, setting `-Drelease` from the command line to activate the release profile and disable the incubator profile to ensure all combinations look right before we do a full test.

Reply:
Thanks @vordimous.

There is usually only one pom per Maven project, and each Maven project has a `src/main` directory, so we'll need to get rid of the poms down in `src/main/docker/incubator` and `src/main/docker/release`.

Instead I think we can move the majority of the content of those poms into corresponding incubator and release profiles of the `docker-image` pom.

Also, I realized that docker build only lets files be copied from a directory local to the `Dockerfile`, so to avoid duplication of `zpm.json.template` and `zilla.properties` for each image, we might need to have more than one `Dockerfile` in the `release` directory, such as `Dockerfile` and `Dockerfile-alpine` perhaps.

I'll look into updating the pom and also see if we can override the name of the file used for `Dockerfile` in the `docker` maven plugin.

Reply:
> There is usually only one pom per Maven project, and each Maven project has a src/main directory, so we'll need to get rid of the poms down in src/main/docker/incubator and src/main/docker/release.

Yeah I was fairly certain you didn't want me to turn those into maven projects to have the child inheritance. I wasn't sure if there was a different way to do it or not. 

> Also, I realized that docker build only lets files be copied from a directory local to the Dockerfile, so to avoid duplication of zpm.json.template and zilla.properties for each image, we might need to have more than one Dockerfile in the release directory, such as Dockerfile and Dockerfile-alpine perhaps.

The plug-in does support filename `dockerFile` config, so we can use alpine.Dockerfile per the [naming convention](https://docs.docker.com/build/building/packaging/#filename)